rs/state_layout/src/state_layout/tests.rs

```diff
@@ -571,4 +595,51 @@ fn read_back_checkpoint_directory_names(heights in random_sorted_unique_heights(
     });
 }
 
+#[test]
+fn read_back_canister_snapshot_ids(mut snapshot_ids in random_unique_snapshot_ids(10, 10, 10)) {
+    let tmp = tmpdir("canister");
```

Comment:
By convention in these tests, `tmpdir("canister")` seems to be only used when we want to create a `CansiterLayout` from it.